.NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseDatePeriodExtractor.cs

```diff
@@ -682,5 +726,63 @@ private bool IsDateRelativeToNowOrToday(ExtractResult er)
 
             return false;
         }
+
+        // Matches "within (the next)?" part (in beforeStr or afterStr) in "within Days/Weeks/Months/Years"
+        private void MatchWithinNextPrexixRegex(string text, Token duration, out int startToken, out int endToken, out bool matchSuccess, bool inPrefix)
```

Comment:
This should preferably return some structure, instead of multiple 'out' parameters.

Reply:
ok, modified to return directly a Token